src/tokens/TokenBalanceItem.tsx

```diff
@@ -0,0 +1,118 @@
+import BigNumber from 'bignumber.js'
+import React from 'react'
+import { Trans } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import { formatValueToDisplay } from 'src/components/TokenDisplay'
+import TokenIcon from 'src/components/TokenIcon'
+import Touchable from 'src/components/Touchable'
+import { getLocalCurrencySymbol, usdToLocalCurrencyRateSelector } from 'src/localCurrency/selectors'
+import useSelector from 'src/redux/useSelector'
+import colors from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import { TokenBalance } from 'src/tokens/slice'
+
+export const TokenBalanceItem = ({ token }: { token: TokenBalance }) => {
+  const localCurrencyExchangeRate = useSelector(usdToLocalCurrencyRateSelector)
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol)
+  const amountInUsd = token?.usdPrice?.multipliedBy(token.balance)
+  const amountInLocalCurrency = new BigNumber(localCurrencyExchangeRate ?? 0).multipliedBy(
+    amountInUsd ?? 0
+  )
+  const showAmount = amountInLocalCurrency.isGreaterThan(0)
+
+  return (
+    <Touchable>
+      <View style={styles.container}>
+        <TokenIcon token={token} viewStyle={styles.marginRight} />
+        <View style={styles.textContainer}>
+          <View style={styles.line}>
+            <Text numberOfLines={1} style={styles.label}>
+              {token.name}
+            </Text>
+            {/* Consider moving formatValueToDisplay out of TokenDisplay */}
+            <Text numberOfLines={1} style={styles.amount}>
```

Comment:
oh i see Satish commented something similar. i guess my main concern is that the logic for displaying token prices / rounding / handling stale and null prices should be handled only once, so if it doesn't make sense to use an existing component, then this should be pulled out to a new component

Reply:
Pulled out the display logic into it's own component that just takes a token balance in https://github.com/valora-inc/wallet/pull/4207/commits/3dea146417ce7acd02511bf7d20d127f72c1faa5. For now it's living in the same file, but it can be moved in the future if needed.